lib/classifier-reborn/bayes.rb

```diff
@@ -112,18 +116,25 @@ def untrain(category, text)
     def classifications(text)
       score = {}
       word_hash = Hasher.word_hash(text, @language, @enable_stemmer)
-      category_keys.each do |category|
-        score[category.to_s] = 0
-        total = (@backend.category_word_count(category) || 1).to_f
-        word_hash.each do |word, _count|
-          s = @backend.word_in_category?(category, word) ? @backend.category_word_frequency(category, word) : 0.1
-          score[category.to_s] += Math.log(s / total)
+      if word_hash.empty?
+        category_keys.each do |category|
+          score[category.to_s] = Float::INFINITY
+        end
+        score
+      else
+        category_keys.each do |category|
+          score[category.to_s] = 0
+          total = (@backend.category_word_count(category) || 1).to_f
+          word_hash.each do |word, _count|
+            s = @backend.word_in_category?(category, word) ? @backend.category_word_frequency(category, word) : 0.1
+            score[category.to_s] += Math.log(s / total)
+          end
+          # now add prior probability for the category
+          s = @backend.category_has_trainings?(category) ? @backend.category_training_count(category) : 0.1
+          score[category.to_s] += Math.log(s / @backend.total_trainings.to_f)
         end
-        # now add prior probability for the category
-        s = @backend.category_has_trainings?(category) ? @backend.category_training_count(category) : 0.1
-        score[category.to_s] += Math.log(s / @backend.total_trainings.to_f)
+        score
```

Comment:
Since you did not used `return` keyword in the `if` section of the `word_hash.empty?` clause and wrapped the remaining code in the `else` clause, I would encourage to remove implicit return of `score` from both the clauses and return it outside the conditional as the last line.

Alternatively, explicit return in the special case and not wrapping rest of the code in `else` clause would reduce a level of nesting though, while gaining the same effect.